### PR TITLE
fix(osc): osc log is not show totally and  flow task is done unnormally

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
module-online schema change
#### What this PR does / why we need it:
1. after`OnlineSchemaChangeFlowableTask` started and executing,  thread  should be wait unit all osc schedule task completed, on `onProgressUpdate` method persist to update task status,   thread executing in `onProgressUpdate` will notify the task started thread.
2. when task percentage or status is changed, then update task
3. fix finalTableDdl for alter sql type

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java [hunk 1]
```diff
@@ -79,14 +79,17 @@ public class OnlineSchemaChangeFlowableTask extends BaseODCFlowTaskDelegate<Void
     @Autowired
     private OrganizationService organizationService;
 
+    private final Object LOCK = new Object();
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java:82]: lock, LOCK is only for static field

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java:82]: instance lock is only monitor current object,  every osc task relational a new OnlineSchemaChangeFlowableTask object

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java [hunk 2]
```diff
@@ -117,6 +120,12 @@ protected Void start(Long taskId, TaskService taskService, DelegateExecution exe
                 taskHandler.start(scheduleId, tasks.get(0).getId());
                 log.info("Successfully start schedule task with id={}", tasks.get(0).getId());
             }
+            synchronized (LOCK) {
+                log.info("Wait task {} to be terminated.", taskId);
+                LOCK.wait();
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java]: we set `oscTaskExpiredAfterSeconds` default value is 5*24*3600 seconds in DefaultOnlineSchemaChangeTaskHandler, which represent osc task will be canceled automatic if it's not completed

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java (hunk 1 above)
Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java:82]: I mean that 'LOCK' this upper case name is usually used for static field, nor for local field.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OnlineSchemaChangeFlowableTask.java]: we can not rely on this timeout settings~

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**